### PR TITLE
Add API authorization and user updates to T&T Server

### PR DESCRIPTION
**Pull request by delventhalz**
Adds an authorization route and update users route , along with corresponding middleware and database API updates. This completes the server-side requirements for [STL-627](https://jira.hyperledger.org/browse/STL-627)

To test:
```
$ cd sawtooth-core/families/track_and_trade/server
$ npm install
$ npm run init 
$ npm start
```
You should now be able to send the following requests (all request and response bodies are JSON):

### POST /users
_request:_
```
{
    "username": <string>,
    "password": <string>,
    "email": <string, must be a valid email>,
    "publicKey": <string>,
    "privateKey": <string>
}
```

_response:_
```
{
    "authorization": <string, JWT auth token>,
    "signing": <private key>
}
```

### POST /authorization
_request:_
```
{
    "username": <string>,
    "password": <string>
}
```

_response:_
```
{
    "authorization": <string, JWT auth token>,
    "signing": <private key>
}
```

### PATCH /users
Must add the JWT token to the `Authorization` header of this request.
_request_ (one or more of these keys):
```
{
    "username": <string>,
    "password": <string>,
    "email": <string, must be a valid email>,
    "privateKey": <string>
}
```

_response:_
```
{
    "username": <string>,
    "email": <string, must be a valid email>,
    "publicKey": <string>,
    "privateKey": <string>
}
```

If you're feeling adventurous, you can go to `localhost:8080` in your browser to explore the data you've created.

### CODE REVIEW

File: families/track_and_trade/server/api/users.js [hunk 1]
```diff
@@ -27,9 +27,27 @@ const create = user => {
       return db.insert(_.assign({}, user, {password: hashed}))
         .catch(err => { throw new BadRequest(err.message) })
     })
-    .then(() => _.omit(user, 'password'))
+    .then(() => auth.createToken(user.publicKey))
+    .then(token => ({
+      authorization: token,
+      signing: user.privateKey
```
Comment (peterschwarz) [inline families/track_and_trade/server/api/users.js]: Is this the encrypted private key?

Comment (delventhalz) [reply families/track_and_trade/server/api/users.js]: Yes. The encryption is 100% client-side though. You could pass the server a plain text private key, and it wouldn't know the difference. It just takes whatever you give it, and stores it.

In this case, the user just sent in the private key, so sending it back is probably redundant, but it keeps it consistent with the response from the auth route.

Comment (peterschwarz) [reply families/track_and_trade/server/api/users.js]: Would suggest naming the field `encryptedSigningKey`, `signing` doesn't really give much information about what the field is, then.

Comment (peterschwarz) [reply families/track_and_trade/server/api/users.js]: Perhaps the same should be said for all the `privateKey` fields.  Making the name of the field `encryptedPrivateKey` will at least give a hint to the user not to send the plain-text private key over the wire.

Comment (delventhalz) [reply families/track_and_trade/server/api/users.js]: This is a good point. I will do some more thinking about what those names should be.

**Comment by nick-drozd** [comment]
I'm consistently getting this:
```
$ npm start

> tnt_server@0.0.0 start /Users/drozd/project/sawtooth-core/families/track_and_trade/server
> rethinkdb & nodemon index.js

Directory 'rethinkdb_data' is already in use, perhaps another instance of rethinkdb is using it.
[nodemon] 1.11.0
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `node index.js`
module.js:487
    throw err;
    ^

Error: Cannot find module 'jsonwebtoken'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/drozd/project/sawtooth-core/families/track_and_trade/server/api/auth.js:22:13)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/drozd/project/sawtooth-core/families/track_and_trade/server/api/index.js:24:14)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
[nodemon] app crashed - waiting for file changes before starting...
```
When I control-c out of `npm run init` and `npm start`, it leaves three `rethinkdb` processes running.

**Comment by delventhalz** [comment]
@nick-drozd The new init script was bad. It is fixed now and shouldn't leave rethink running. Make sure you `npm install` again, as `jsonwebtoken` is a new dependency.

**Pull request merged**